tests/WE2E/run_WE2E_tests.sh

```diff
@@ -430,10 +430,24 @@ elif [ -n "${tests_file}" ] || [ -n "${test_type}" ] ; then
   # one managed in the repo
 
   if [ -n "${test_type}" ] ; then
-    # Check for a pre-defined set. It could be machine dependent or not.
-    user_spec_tests_fp=${scrfunc_dir}/machine_suites/${test_type}.${machine}
+    # Check for a pre-defined set. It could be machine dependent or has the mode
+    # (community or nco), or default
+    user_spec_tests_fp=${scrfunc_dir}/machine_suites/${test_type}.${machine}.nco
     if [ ! -f ${user_spec_tests_fp} ]; then
-        user_spec_tests_fp=${scrfunc_dir}/machine_suites/${test_type}
+        user_spec_tests_fp=${scrfunc_dir}/machine_suites/${test_type}.${machine}.com
+        if [ ! -f ${user_spec_tests_fp} ]; then
+            user_spec_tests_fp=${scrfunc_dir}/machine_suites/${test_type}.${machine}.${compiler}
+            if [ ! -f ${user_spec_tests_fp} ]; then
+                user_spec_tests_fp=${scrfunc_dir}/machine_suites/${test_type}.${machine}
+                if [ ! -f ${user_spec_tests_fp} ]; then
+                    user_spec_tests_fp=${scrfunc_dir}/machine_suites/${test_type}
+                fi
+            fi
+        else
+            run_envir=${run_envir:-"community"}
+        fi
+    else
+        run_envir=${run_envir:-"nco"}
```

Comment:
@danielabdi-noaa I hadn't checked on this PR before it was merged because I thought it would only change NCO mode running, but this seems to have introduced the additional change that the fundamental test suite on Hera will *always* be run in NCO mode. Was this an intentional change? If so, what was the reasoning?

Reply:
@mkavulich Yes, it is intentional. The goal for the fundamental test suites has shifted with this PR, now it is more about expanded test coverage and variety. I chose Hera for NCO mode so that it would be tested with the Github Actions CI as well, which runs only on Hera/Jet. The fundamental test suite is now running different test cases on each platform, each compiler gnu/intel. Last time I check it was close to being "comprehensive" with 45 test cases or so.

The logic in this code is kind of convoluted and if you find a better way to rename the files while keeping the flexibility, it would be great!

Reply:
Thanks for getting back to me so quickly. I may try to bring this up for discussion at the next code management meeting, because this feels like it may be confusing to developers testing their changes on a specific machine. Although perhaps a better topic for discussion would be the broader "which WE2E tests should developers be running to test their code prior to opening a PR", since this is not yet covered in the contributors guide, and clearly I didn't even understand what tests were being run when I ran the script before now!